tests/by-util/test_split.rs

```diff
@@ -738,6 +738,158 @@ fn test_hex_suffix() {
     assert_eq!(at.read("x0c"), "");
 }
 
+#[test]
+fn test_numeric_suffix_no_equal() {
+    new_ucmd!()
+        .args(&["-n", "4", "--numeric-suffixes", "9", "threebytes.txt"])
+        .fails()
+        .stderr_contains("split: cannot open '9' for reading: No such file or directory");
+}
+
+#[test]
+fn test_hex_suffix_no_equal() {
+    new_ucmd!()
+        .args(&["-n", "4", "--hex-suffixes", "9", "threebytes.txt"])
+        .fails()
+        .stderr_contains("split: cannot open '9' for reading: No such file or directory");
+}
+
+/// Test for short numeric suffix not having any value
+#[test]
+fn test_short_numeric_suffix_no_value() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    ucmd.args(&["-n", "4", "-d", "threebytes.txt"])
+        .succeeds()
+        .no_stdout()
+        .no_stderr();
+    assert_eq!(at.read("x00"), "a");
+    assert_eq!(at.read("x01"), "b");
+    assert_eq!(at.read("x02"), "c");
+    assert_eq!(at.read("x03"), "");
+}
+
+/// Test for long numeric suffix not having any value
+#[test]
+fn test_numeric_suffix_no_value() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    ucmd.args(&["-n", "4", "--numeric-suffixes", "threebytes.txt"])
+        .succeeds()
+        .no_stdout()
+        .no_stderr();
+    assert_eq!(at.read("x00"), "a");
+    assert_eq!(at.read("x01"), "b");
+    assert_eq!(at.read("x02"), "c");
+    assert_eq!(at.read("x03"), "");
+}
+
+/// Test for short hex suffix not having any value
+#[test]
+fn test_short_hex_suffix_no_value() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    ucmd.args(&["-n", "4", "-x", "threebytes.txt"])
+        .succeeds()
+        .no_stdout()
+        .no_stderr();
+    assert_eq!(at.read("x00"), "a");
+    assert_eq!(at.read("x01"), "b");
+    assert_eq!(at.read("x02"), "c");
+    assert_eq!(at.read("x03"), "");
+}
+
+/// Test for long hex suffix not having any value
+#[test]
+fn test_hex_suffix_no_value() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    ucmd.args(&["-n", "4", "--hex-suffixes", "threebytes.txt"])
```

Comment:
I think we should test with a larger file here so that we can verify that hex is used (same for the numeric suffix tests).

Reply:
yep, will do